pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -99,6 +99,36 @@ func Convert_v1_Disk_To_api_Disk(diskDevice *v1.Disk, disk *Disk, devicePerBus m
 	return nil
 }
 
+func setDriverCacheMode(disk *Disk, mode v1.DriverCache) error {
+	supportDirectIO := true
+	if mode == "" || mode == v1.CacheNone {
+		f, err := os.OpenFile(disk.Source.File, syscall.O_RDONLY|syscall.O_DIRECT, 0)
+		f.Close()
+		if err != nil && !os.IsNotExist(err) {
+			supportDirectIO = false
+		}
+	}
+
+	// if user set a cache mode = 'none' and fs does not support direct I/O then return an error
+	if mode == v1.CacheNone && !supportDirectIO {
+		return fmt.Errorf("Unable to use '%s' cache mode, file system where %s is stored does not support direct I/O", mode, disk.Source.File)
+	}
+
+	if mode == "" {
+		// if user did not set a cache mode and fs supports direct I/O then set cache = 'none'
+		// else set cache = 'writethrough'
+		if supportDirectIO {
+			disk.Driver.Cache = string(v1.CacheNone)
+		} else {
+			disk.Driver.Cache = string(v1.CacheWriteThrough)
```

Comment:
we should at least log here, that we are falling back to writethrough because the fs doesn't support o_direct.

Reply:
logs are added